airflow/contrib/hooks/datadog_hook.py

```diff
@@ -47,26 +47,15 @@ def __init__(self, datadog_conn_id='datadog_default'):
         # for all metric submissions.
         self.host = conn.host
 
-        if self.api_key is None:
-            raise AirflowException("api_key must be specified in the "
-                                   "Datadog connection details")
-        if self.app_key is None:
-            raise AirflowException("app_key must be specified in the "
-                                   "Datadog connection details")
```

Comment:
In absence of this code block when a user forgets to have `api_key` in Connection extras, it won't give a useful error. Any reasoning behind removing this?

Reply:
In some cases, the app key isn't required (from https://docs.datadoghq.com/api/?lang=python#authentication):

> Requests that write data require reporting access and require an API key. Requests that read data require full access and also require an application key.

I'll update so that the api key is required and the app key isn't.